emcc.py

```diff
@@ -1927,11 +1927,17 @@ def phase_linker_setup(options, state, newargs):
   if '_main' in settings.EXPORTED_FUNCTIONS:
     settings.EXPORT_IF_DEFINED.append('__main_argc_argv')
 
-  # -sASSERTIONS implies basic stack overflow checks, and ASSERTIONS=2
-  # implies full stack overflow checks.
   if settings.ASSERTIONS:
-    # However, we don't set this default in PURE_WASI, or when we are linking without standard
-    # libraries because STACK_OVERFLOW_CHECK depends on emscripten_stack_get_end which is defined
+    # Exceptions are thrown with a stack trace by default when ASSERTIONS is
+    # set and when building with either -fexceptions or -fwasm-exceptions.
+    if 'EXCEPTION_STACK_TRACES' in user_settings and not settings.EXCEPTION_STACK_TRACES:
+      exit_with_error('EXCEPTION_STACK_TRACES cannot be disabled when ASSERTIONS are enabled')
+    default_setting('EXCEPTION_STACK_TRACES', not settings.DISABLE_EXCEPTION_CATCHING or settings.WASM_EXCEPTIONS)
```

Comment:
Can you leave this as written below: 

```
if not settings.DISABLE_EXCEPTION_CATCHING or settings.WASM_EXCEPTIONS:		
  settings.EXCEPTION_STACK_TRACES = 1
```

Using `default_settings` is for the cases where you want to allow the user to control the settings.. in this case we want to set the setting with prejudice (that warning above basically tells the user we are about to do this).

In fact can you flip the condition to be `if settings.WASM_EXCEPTIONS or not settings.DISABLE_EXCEPTION_CATCHING:`  (seems a little more readable that way around).

Reply:
Ah, this was changed with commit 9f93b10b6b9053ad19af4c1436591b34ecbfd807 and I forgot to re-sync that condition during the rebase.

Fixed with commit 5cb4a1849b9f7af39248292c66860d2925fec8e8.